### PR TITLE
Clone and NoDrop.

### PR DESCRIPTION
**Pull request by enkodellc**
Added "data-clone" element.  This could be improved upon with UI changes maintaining the source node in the source tree.
Fixed the "data-nodrop" attribute. This could be improved upon with UI changes not allowing a placeholder to be placed in the destination tree.
I didn't have time to add the tests.
Fix issue with an Angular $watch on the tree ng-model firing twice on every drop: https://github.com/angular-ui-tree/angular-ui-tree/pull/396
data-nodrag can be set to true or false: https://github.com/angular-ui-tree/angular-ui-tree/pull/379

Don't force placeholder width - if you have 2 trees of different sizes the dropping elements are now the correct width: https://github.com/angular-ui-tree/angular-ui-tree/pull/381

Corrected attribute name and fixed typos: https://github.com/angular-ui-tree/angular-ui-tree/pull/410/files

I included other PR's as I needed all of them for my project to work together. Thanks for those who submitted the other noted PR's.

### CODE REVIEW

**Comment by meriturva** [comment]
Please merge this pull request!

**Comment by enkodellc** [comment]
@meriturva  thanks, but I haven't seen much action from this repo... @JimLiu  any response to this PR? Thanks for all your hard work on this directive.

**Comment by meriturva** [comment]
@JimLiu what do you think to extend projects admin to others? just to restart the development and bugfix!....so thanks anyway.

File: README.md [hunk 1]
```diff
@@ -131,6 +131,16 @@ Injecting `ui.tree`, `ui-tree-nodes`, `ui-tree-node`, `ui-tree-handle` to your h
 `ui-tree` is the root scope for a tree
 
 #### Attributes
+##### data-nodrop
+Turn on the ability to prevent dropping of nodes into this tree.
+- `false` (default): turn off
+- `true`: turn on no drop
+
+##### data-clone
```
Comment (Voles) [inline README.md]: What do you think of using `data-clone-enabled`? A little more verbose and in line with `data-drag-enabled`.

File: README.md [hunk 2]
```diff
@@ -131,6 +131,16 @@ Injecting `ui.tree`, `ui-tree-nodes`, `ui-tree-node`, `ui-tree-handle` to your h
 `ui-tree` is the root scope for a tree
 
 #### Attributes
+##### data-nodrop
```
Comment (Voles) [inline README.md]: What do you think of using `data-drop-enabled`?

**Comment by Voles** [comment]
Thanks for the PR! :cake: 
Please let me know what you think about my comments.

**Comment by enkodellc** [comment]
@Voles  I will refacter with your comments tonight. I agree that these are consistent naming conventions with ui-tree. Thanks.

**Comment by enkodellc** [comment]
@Voles I made the changes and added a demo page for the features. Please review and test. Thanks

**Comment by Voles** [comment]
Looks good, thanks for the updates!

**Comment by enkodellc** [comment]
@Voles Thanks,  please review my notes as this closes a few issues...

**Pull request merged**